### ISSUE TITLE
Enhancement Request: Compressor Output integration

### ISSUE THREAD

**Issue by tdlivings** [mackworth/cTiVo#398]
It would be nice to have the cTiVo app recognize and use compressor droplets so we might start to use both eGPU and distributed jobs among Mac clusters. Well, who knows, it doesn't hurt to ask. Maybe an added option for a future pay version.

**Comment by mackworth** [comment]
BTW, I've looked at this request several times over the last year, and keep coming back to "I don't own Compressor, so I can't do it".   I've read the online documentation, and there is a command line interface. But there doesn't seem to be any mechanism to determine when a job is complete (so cTiVo can do its ending stuff, such as merging in captions, commercial chapters, etc). It would be great to know what percent complete a job is, but that one can be faked.

There's a possible looking command: `-checkstream`  but it seems more likely to be a validation. Can you type in `Compressor -help` in Terminal and see if there's any more information than [here](https://support.apple.com/guide/compressor/syntax-cpsr9be73312/mac)?

**Comment by tdlivings** [comment]
Will do. But for now the handbrake profile works in a way that is was hoping to use compressor. I created a custom profile and use a derivative of production proxy 1080.

**Comment by tdlivings** [comment]
compressor -help indicates compressor not a valid command

**Comment by mackworth** [comment]
Assuming you have Compressor installed, you’ll have to find the folder it is installed in. 

Further assuming it’s in Applications, then the command would be 
`/Applications/Compressor.app/Contents/MacOS/compressor -help`

**Comment by tdlivings** [comment]
Usage:  Compressor [Cluster Info] [Batch Specific Info] [Optional Info] [Other Options]

	-computergroup <name> -- name of the Computer Group to use.
--Batch Specific Info:--
	-batchname <name> -- name to be given to the batch.
	-priority <value> -- priority to be given to the batch. Possible values are: low, medium or high 
Job Info: Used when submitting individual source files. Following parameters are repeated to enter multiple job targets in a batch
	-jobpath <url> -- url to source file.
				   -- In case of Image Sequence, URL should be a file URL pointing to directory with image sequence.
				   -- Additional URL query style parameters may be specified to set frameRate (file:///myImageSequenceDir?frameRate=29.97) and audio file (e.g. file:///myImageSequenceDir?audio=/usr/me/myaudiofile.mov). 
	-settingpath <path> -- path to settings file.
	-locationpath <path> -- path to location file.
	-info <xml> -- xml for job info.
	-jobaction <xml> -- xml for job action.
	-scc <url> -- url to scc file for source
	-startoffset <hh:mm:ss;ff> -- time offset from beginning
	-in <hh:mm:ss;ff> -- in time
	-out <hh:mm:ss;ff> -- out time
	-annotations <path> -- path to file to import annotations from; a plist file or a Quicktime movie
	-chapters <path> -- path to file to import chapters from
--Optional Info:--
	-help -- Displays, on stdout, this help information.
	-checkstream <url> -- url to source file to analyze
	-findletterbox <url> -- url to source file to analyze

--Batch Monitoring Info:--
Actions on Job:
	-monitor -- monitor the job or batch specified by jobid or batchid.
	-kill -- kill the job or batch specified by jobid or batchid.
	-pause -- pause the job or batch specified by jobid or batchid.
	-resume -- resume previously paused job or batch specified by jobid or batchid.
Optional Info:
	-jobid <id> -- unique id of the job usually obtained when job was submitted.
	-batchid <id> -- unique id of the batch usually obtained when job was submitted.
	-query <seconds> -- The value in seconds, specifies how often to query the cluster for job status.
	-timeout <seconds> -- the timeOut value, in seconds, specifies when to quit the process.
	-once -- show job status only once and quit the process.

--Sharing Related Options:--
	-resetBackgroundProcessing [cancelJobs] -- Restart all processes used in background processing, and optionally cancel all queued jobs.

	-repairCompressor -- Repair Compressor config files and restart all processes used in background processing.

	-sharing <on/off>  -- Turn sharing of this computer on or off.

	-requiresPassword [password] -- Sharing of this computer requires specified password. Computer must not be busy processing jobs when you set the password.

	-noPassword  -- Turn off the password requirement for sharing this computer.

	-instances <number>  -- Enables additional Compressor instances. 

	-networkInterface <bsdname>  -- Specify which network interface to use. If "all" is specified for <bsdname>, all available network interfaces are used.

	-portRange <startNumber> <count>  -- Defines what port range use, using start number specifying how many ports to use.

--File Modification Options (all parameters EXCEPT -jobpath and -locationpath are ignored):--
	-relabelaudiotracks <layout[1] layout[2]... layout[N]
		Supported values:
			Ls : Left Surround
			R : Right
			C : Center
			Rs : Right Surround
			Lt : Left Total
			L : Left
			Rt : Right Total
			LFE : LFE Screen
			Lc : Left Center
			Rls : Rear Surround Left
			mono : Mono
			LtRt : Matrix Stereo (Lt Rt)
			Rc : Right Center
			stereo : Stereo (L R)
			Rrs : Rear Surround Right
        -jobpath <url> -- url to source file. - Must be a QuickTime Movie file
        --Optional Info:--
            -renametrackswithlayouts (Optional, rename the tracks with the new channel layouts)
            -locationpath <path> -- path to location file.  Modified movie will be saved here.  If unspecified, changes will be saved in place, overwriting the original file.
    -relabelcolorspace <primaries> <transfer> <matrix>  (these are integer values corresponding to the NCLC standard
	    -jobpath <url> -- url to source file. - Must be a QuickTime Movie file
	    --Optional Info:--
            -locationpath <path> -- path to location file.  Modified movie will be saved here.  If unspecified, changes will be saved in place, overwriting the original file.

thomaslivings@Thomas-MacBook-Pro Compressor %

**Comment by mackworth** [comment]
Hey!  Looks like this may work.  Can you do the following command in Terminal?
	`ls ~/Library/Application\ Support/Compressor/Settings/`

**Comment by tdlivings** [comment]
do I need CTiVo first, I don't have it on this unit?

**Comment by mackworth** [comment]
nope; not yet. I'm going to send you a couple of Compressor-only batch files to run in Terminal, and the results will then inform what I do for a cTiVo-specific version.

Also, can you give me that pathname of a video you have, say, in ~/Movies?

**Comment by tdlivings** [comment]
setting-group-metadata.group

**Comment by tdlivings** [comment]
/Users/thomaslivings/Movies/TV/Media.localized/TV Shows/Cold Case/0513 Cold Case - Spiders.m4v

**Comment by mackworth** [comment]
hmm, that's not what Apple says in their docs.  I assume you have a custom Compressor settings?

Maybe try 
 `ls -l ~/Library/Application\ Support/Compressor/Settings/setting-group-metadata.group/`
and post the results.

 if it says 'Not a directory' then 
 in Finder, type Cmd-Shift-G and `~/Library/Application Support/Compressor/Settings`
and drag the `setting-group-metadata.group` file into an entry here.

**Comment by tdlivings** [comment]
ls: /Users/thomaslivings/Library/Application Support/Compressor/Settings/setting-group-metadata.group/: Not a directory

**Comment by tdlivings** [comment]
TiVo HD.compressorsetting

**Comment by tdlivings** [comment]
/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting

**Comment by mackworth** [comment]
So, you created a setting, and it appeared there? Good. 

Now, copy/paste the following lines into Terminal, and paste the results here.

I don't expect the second and third one to work, but if it does, you may need to type Ctrl-C after a few seconds to stop it.

After the batch runs, look in Compressor and confirm one or more jobs are running (and you can then kill them).  You can then delete any ctivoJunk files in ~/Movies

````
date
/Applications/Compressor.app/Contents/MacOS/Compressor \
    -jobpath ~/Movies/TV/Media.localized/TV Shows/Cold Case/0513 Cold Case\ -\ Spiders.m4v \
    -settingpath ~/Library/Application\ Support/Compressor/Settings/TiVo\ HD.compressorsetting \ 
    -locationpath ~/Movies/ctivoJunk1.m4v
date
/Applications/Compressor.app/Contents/MacOS/Compressor \
    -batchname "My Second Batch" \
    -jobpath ~/Movies/TV/Media.localized/TV Shows/Cold Case/0513 Cold Case\ -\ Spiders.m4v \
    -settingpath ~/Library/Application\ Support/Compressor/Settings/TiVo\ HD.compressorsetting \ 
    -locationpath ~/Movies/ctivoJunk2.m4v
    -monitor
date
/Applications/Compressor.app/Contents/MacOS/Compressor  -monitor  -batchid 1
date
````

**Comment by mackworth** [comment]
Oops. Typo.  Please use fixed version.

**Comment by mackworth** [comment]
Fixed version:
````
date
/Applications/Compressor.app/Contents/MacOS/Compressor \
    -jobpath ~/Movies/TV/Media.localized/TV Shows/Cold Case/0513 Cold Case\ -\ Spiders.m4v \
    -settingpath ~/Library/Application\ Support/Compressor/Settings/TiVo\ HD.compressorsetting \ 
    -locationpath ~/Movies/ctivoJunk1.m4v
date
/Applications/Compressor.app/Contents/MacOS/Compressor \
    -batchname "My Second Batch" \
    -jobpath ~/Movies/TV/Media.localized/TV Shows/Cold Case/0513 Cold Case\ -\ Spiders.m4v \
    -settingpath ~/Library/Application\ Support/Compressor/Settings/TiVo\ HD.compressorsetting \ 
    -locationpath ~/Movies/ctivoJunk2.m4v \
    -monitor
date
/Applications/Compressor.app/Contents/MacOS/Compressor  -monitor  -batchid 1
date
````

**Comment by mackworth** [comment]
Still wrong. sorry
just a second

**Comment by tdlivings** [comment]
/Applications/Compressor.app/Contents/MacOS/Compressor  -monitor  -batchid 1
date
Sat Nov 14 17:05:16 PST 2020
Parameter error: Invalid Job path: /Users/thomaslivings/Movies/TV/Media.localized/TV; Path does not exist
zsh: command not found: -locationpath
Sat Nov 14 17:05:16 PST 2020
Parameter error: Invalid Job path: /Users/thomaslivings/Movies/TV/Media.localized/TV; Path does not exist
zsh: command not found: -locationpath
Sat Nov 14 17:05:17 PST 2020

**Comment by tdlivings** [comment]
going offline for 20 minutes

**Comment by mackworth** [comment]
Okay.  Here's the revised one  (Terminal very picky about spaces

````
date
/Applications/Compressor.app/Contents/MacOS/Compressor \
    -jobpath ~/Movies/TV/Media.localized/TV\ Shows/Cold Case/0513 Cold Case\ -\ Spiders.m4v \
    -settingpath ~/Library/Application\ Support/Compressor/Settings/TiVo\ HD.compressorsetting \
    -locationpath ~/Movies/ctivoJunk1.m4v
date
/Applications/Compressor.app/Contents/MacOS/Compressor \
    -batchname "My Second Batch" \
    -jobpath ~/Movies/TV/Media.localized/TV\ Shows/Cold Case/0513 Cold Case\ -\ Spiders.m4v \
    -settingpath ~/Library/Application\ Support/Compressor/Settings/TiVo\ HD.compressorsetting \
    -locationpath ~/Movies/ctivoJunk2.m4v \
    -monitor
date
/Applications/Compressor.app/Contents/MacOS/Compressor  -monitor  -batchid 1
date
````

**Comment by tdlivings** [comment]
Sat Nov 14 17:25:17 PST 2020
Parameter error: Invalid Job path: /Users/thomaslivings/Movies/TV/Media.localized/TV Shows/Cold; Path does not exist
Sat Nov 14 17:25:17 PST 2020
Parameter error: Invalid Job path: /Users/thomaslivings/Movies/TV/Media.localized/TV Shows/Cold; Path does not exist
Sat Nov 14 17:25:17 PST 2020

**Comment by tdlivings** [comment]
hold on. maybe I can simplify

**Comment by tdlivings** [comment]
/Users/thomaslivings/Movies/TV/Media.localized/TVS/0513ColdCaseSpiders.m4v

**Comment by tdlivings** [comment]
also 

/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVoHD.compressorsetting

**Comment by tdlivings** [comment]
results:

/Applications/Compressor.app/Contents/MacOS/Compressor \
    -jobpath ~/Movies/TV/Media.localized/TVS/0513ColdCaseSpiders.m4v \
    -settingpath ~/Library/Application\ Support/Compressor/Settings/TiVoHD.compressorsetting \
    -locationpath ~/Movies/ctivoJunk1.m4v
date
/Applications/Compressor.app/Contents/MacOS/Compressor \
    -batchname "My Second Batch" \
    -jobpath ~/Movies/TV/Media.localized/TVS/0513ColdCaseSpiders.m4v\
    -settingpath ~/Library/Application\ Support/Compressor/Settings/TiVoHD.compressorsetting \
    -locationpath ~/Movies/ctivoJunk2.m4v \
    -monitor
date
/Applications/Compressor.app/Contents/MacOS/Compressor  -monitor  -batchid 1
date
Sat Nov 14 17:35:13 PST 2020
2020-11-14 17:35:14.677 Compressor[2571:214330] owner is: thomaslivings
2020-11-14 17:35:14.696 Compressor[2571:214330] <jobID B068B027-6CBE-41D2-B96F-4153ABF8FE0E /> <batchID 320263D5-A5FF-4F65-9EC3-CFE8935B46E7 />
Sat Nov 14 17:35:14 PST 2020
2020-11-14 17:35:15.563 Compressor[2576:214492] owner is: thomaslivings
2020-11-14 17:35:15.580 Compressor[2576:214492] <jobID 8B99DC43-BED5-4FFD-AA24-FD24ABCF6AB4 /> <batchID 934A93AF-1564-4979-8A1F-B026BD1D70BA />
Sat Nov 14 17:35:15 PST 2020
Sat Nov 14 17:35:25 PST 2020

**Comment by tdlivings** [comment]
Running batch

**Comment by tdlivings** [comment]
Ran two batches successfully

**Comment by tdlivings** [comment]
elapsed time per batch 2 minutes

**Comment by mackworth** [comment]
Great. So we can launch a job, now we need to monitor it. One more request, and I should be able to build the cTiVo batch file. 

Run the first compressor line again,
````
/Applications/Compressor.app/Contents/MacOS/Compressor \
    -jobpath ~/Movies/TV/Media.localized/TVS/0513ColdCaseSpiders.m4v \
    -settingpath ~/Library/Application\ Support/Compressor/Settings/TiVoHD.compressorsetting \
    -locationpath ~/Movies/ctivoJunk1.m4v
date
````

 then copy/paste the resulting batch id (e.g. 934...0AB above) into this line:
```` Applications/Compressor.app/Contents/MacOS/Compressor  -monitor  -batchid BATCHIDHERE
````

**Comment by tdlivings** [comment]
sorry asleep at the wheel

**Comment by tdlivings** [comment]
failed the first time, I had to delete the old file, no error message. After I deleted the existing file from previous job the batch launched successfully

**Comment by tdlivings** [comment]
2020-11-14 18:45:28.337 Compressor[2824:239654] owner is: thomaslivings
2020-11-14 18:45:28.353 Compressor[2824:239654] <jobID 197B1255-06F8-481D-B6B8-04DD18D18117 /> <batchID 0E847C38-4CF5-4BBE-9196-384FF2820CC8 />

**Comment by tdlivings** [comment]
It is interesting that it will not over write the existing file if the same name, I just get a softfail

**Comment by mackworth** [comment]
Deletion doesn’t matter, cTiVo will handle that. 

Sigh, github hid the last part of my message above. I fixed that one, but here it is again:

Run the first compressor line again,
````
/Applications/Compressor.app/Contents/MacOS/Compressor \
    -jobpath ~/Movies/TV/Media.localized/TVS/0513ColdCaseSpiders.m4v \
    -settingpath ~/Library/Application\ Support/Compressor/Settings/TiVoHD.compressorsetting \
    -locationpath ~/Movies/ctivoJunk1.m4v
date
````

 then copy/paste the resulting batch id (e.g. 934...0AB above) into this line:

````
/Applications/Compressor.app/Contents/MacOS/Compressor  -monitor  -batchid BATCHIDHERE
````

I need to know what the output if the monitor command looks like.

**Comment by tdlivings** [comment]
thomaslivings@Thomas-MacBook-Pro ~ % /Applications/Compressor.app/Contents/MacOS/Compressor  -monitor  -batchid F8265407-1F41-4AEE-A9EC-5B2DA0924FCE
<batchStatus name="Commandline Batch" submissionTime="11/14/20, 6:55:49 PM" sentBy="thomaslivings" timeElapsed="2 minute(s)" timeRemaining="0" timeElapsedSeconds="122" timeRemainingSeconds="0" percentComplete="100" resumePercentComplete="100" status="Successful" batchid="F8265407-1F41-4AEE-A9EC-5B2DA0924FCE" /batchStatus>
<jobStatus name="0513ColdCaseSpiders.m4v" submissionTime="11/14/20, 6:55:49 PM" sentBy="thomaslivings" jobType="" priority="HighPriority" timeElapsed="2 minute(s)" timeRemaining="0" timeElapsedSeconds="122" timeRemainingSeconds="0" percentComplete="100" resumePercentComplete="100" status="Successful" jobid="14C31513-B7E0-4BF8-85C0-957F4A0CB923" batchid="F8265407-1F41-4AEE-A9EC-5B2DA0924FCE" /jobStatus>

**Comment by tdlivings** [comment]
is that right?

**Comment by mackworth** [comment]
That’s perfect, although it looks like it was run after the job was fully complete. Nonetheless, I have what I need for now.

**Comment by mackworth** [comment]
Okay, let's try this out.   Attached is a zip with two files. Download it and expand them. 
Then move the compressorBatch.sh file to someplace convenient. 
Go into Terminal and type `chmod +x ` and then drag the compressorBatch.sh onto the Terminal window and hit return.
Then in cTiVo, `File>Import Formats>` and select the compressor.plist file.  
Go into Edit Formats, select the Compressor Format 
Modify the `Encoder to use` field to point to wherever you put your compressorBatch.sh file. Make sure text turns black.

Then in Advanced Prefs, set Download and Task logging to Detail.  
In Compressor, rename your settings from `TiVoHD` back to `TiVo HD`.  (default name in script)

Then try it out. On your TiVo, you might want to record a VERY short video to test with. 
     Change channels (so you don't get the last half hour cached), and hit Record; wait about 30 seconds and hit Stop. 
Change your default Format to Compressor.  
Turn off captions and commercialling initially. 
Download the show you just recorded. 

Given the persnicketiness of shell scripts and my inability to test with a real Compressor, it will undoubtable fail. 
If so, do the "Show logs in Finder", right-click to compress the latest log, and post it.

If by some odd chance, it works (or after we make it work), then please experiment.  

A few I can think of are:
Turn on captions and comskip and see if they work (may need a longer video).
Try a different settings file. Just add `-settings "new settings"` or whatever you called it to `Other Options for Encoder`.  Please see if space works ok.
Try two at a time. Be sure the first is very short.
Move your copy of Compressor to a different folder (e.g. `/Applications/Video Test`, then add  `-appdir  "/Applications/Video Test"` to  `Other Options for Encoder`.
Try your multiple CPU scenario!

There's still one serious problem I know of: If you allow more than one `Concurrent Encoders` entry in Adv Prefs, then later ones will time out if the first one takes more than 2 minutes to encode. Assuming we get this to work, I'll need to think more about this one.

[compressor.zip](https://github.com/mackworth/cTiVo/files/5543133/compressor.zip)

**Comment by tdlivings** [comment]
I apologize but I am offline for work until Monday night. I will be close to a dev box Monday night  I am excited to try it out.

**Comment by mackworth** [comment]
No prob. Talk to you then.

**Comment by tdlivings** [comment]
It keeps failing

**Comment by tdlivings** [comment]
[com.cTiVo.cTiVo 2020-11-17--03-57-12-132.log](https://github.com/mackworth/cTiVo/files/5551216/com.cTiVo.cTiVo.2020-11-17--03-57-12-132.log)

**Comment by tdlivings** [comment]
it downloads but then fails to send to compressor and the program keeps downloading and failing

**Comment by tdlivings** [comment]
[Terminal Saved Output.txt](https://github.com/mackworth/cTiVo/files/5551242/Terminal.Saved.Output.txt)
 maybe the chmod didn't take?

**Comment by tdlivings** [comment]
[Terminal Saved Output.txt](https://github.com/mackworth/cTiVo/files/5551264/Terminal.Saved.Output.txt)
 ran it again, still fails after download stage. 
[com.cTiVo.cTiVo 2020-11-17--03-57-12-132.log](https://github.com/mackworth/cTiVo/files/5551266/com.cTiVo.cTiVo.2020-11-17--03-57-12-132.log)

**Comment by mackworth** [comment]
Argh.  I'm so sorry. 

I sent you the wrong shell file.  Compressor.sh is a "Compressor simulator" that I created to debug my CompressorBatch.sh.

Here's the compressorBatch.sh file. Make sure that your Compressor Format is pointing to this file, not Compressor.sh

(BTW, on the `chmod +x`, don't type Return after `chmod +x` before dragging in  the file. You want the dragged file to complete that line.) That doesn't look like it was necessary though.

[compressorBatch.sh.zip](https://github.com/mackworth/cTiVo/files/5551537/compressorBatch.sh.zip)

**Comment by tdlivings** [comment]
When I imported it and the edited the sh address the compressor name grey's out. I had to duplicate it with compressor-x to get it listed in the drop down. 

[com.cTiVo.cTiVo 2020-11-17--06-34-01-244.log](https://github.com/mackworth/cTiVo/files/5551736/com.cTiVo.cTiVo.2020-11-17--06-34-01-244.log)
 failed

**Comment by mackworth** [comment]
It says it can’t find the settings file at ~/Library/Application\ Support/Compressor/Settings/TiVo HD.compressorsetting

If it’s actually there, then I have a space quoting problem.

**Comment by mackworth** [comment]
No need to reimport the plist again

**Comment by tdlivings** [comment]
<img width="794" alt="Screen Shot 2020-11-17 at 07 28 57" src="https://user-images.githubusercontent.com/22117759/99409847-aaca5100-28a6-11eb-8b7f-c5a9a0b0c21f.png">

**Comment by mackworth** [comment]
I did a full audit on space handling, so this should be better on that and some more diagnostics.

Thanks again for your patience here for remote debugging.

[compressorBatch.sh.zip](https://github.com/mackworth/cTiVo/files/5555589/compressorBatch.sh.zip)

**Comment by tdlivings** [comment]
I think it is you I should thank. My job is brutal and I am offline for 2 more days lol. This is how it goes. But it is better than being unemployed. More Wednesday night.

**Comment by mackworth** [comment]
Thanks.  TTYT

**Comment by tdlivings** [comment]
OK, I think we are making progress. It downloaded and transitioned to encoding. However, no job is listed in compressor and no progress is happening, and then it starts downloading again.

[com.cTiVo.cTiVo 2020-11-19--03-59-48-762.log](https://github.com/mackworth/cTiVo/files/5564288/com.cTiVo.cTiVo.2020-11-19--03-59-48-762.log)

**Comment by mackworth** [comment]
Hmmm. Because the encoding launched “successfully”, cTiVo is not showing the text file. Can you turn on “don’t delete tmp files” in adv prefs, then download it again? (You can then turn off the option.) Then copy/paste the following in Terminal: 

````
cd /Users/thomaslivings/Downloads

mv '/var/folders/lv/tf9hpmz91kn14zxx2s643c880000gn/T/ctivo/bufferIce Road Truckers-Journey to the End of the Earth.mpg' journey.mpg

./compressor/compressorBatch.sh -i journey.mpg journey.MP4
````

It’s copying the raw video file into your download folder, so we can then run the batch file without cTiVo being involved. 

And let me know what it says...

**Comment by tdlivings** [comment]
the raw file is waiting in downloads folder

**Comment by tdlivings** [comment]
Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      journey.mpg
Output file:     journey.MP4

but nothing launched in compressor

**Comment by mackworth** [comment]
Are you sure `Downloads/Compressor/compresdorbatch.sh` is the latest (the one with the space audit)? The output should start with “version 0.1”

**Comment by tdlivings** [comment]
yes

**Comment by tdlivings** [comment]
<img width="244" alt="Screen Shot 2020-11-18 at 9 35 13 PM" src="https://user-images.githubusercontent.com/22117759/99625855-f5061c00-29e5-11eb-86a2-a1a794b14fb0.png">

**Comment by tdlivings** [comment]
you have the location wrong in the script

**Comment by tdlivings** [comment]
./compressor/compressorBatch.sh

**Comment by mackworth** [comment]
Not understanding. if you’re in Downloads, then ./compressor/compressorBatch.sh should be right?  If not, can you run it with the right one?

**Comment by tdlivings** [comment]
Oh yes never mind

**Comment by mackworth** [comment]
If you do `cat compressor/compressorBatch.sh`, you should see the `echo “version 0.1”` line near the top.

**Comment by tdlivings** [comment]
EOF
exit 1
}
echo "Version 0.1"


Looks like the new one

**Comment by tdlivings** [comment]
mv '/var/folders/lv/tf9hpmz91kn14zxx2s643c880000gn/T/ctivo/bufferIce Road Truckers-Journey to the End of the Earth.mpg' journey.mpg

./compressor/compressorBatch.sh -i journey.mpg journey.MP4
mv: rename /var/folders/lv/tf9hpmz91kn14zxx2s643c880000gn/T/ctivo/bufferIce Road Truckers-Journey to the End of the Earth.mpg to journey.mpg: No such file or directory
Version 0.1

Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      journey.mpg
Output file:     journey.MP4

**Comment by mackworth** [comment]
Ah!  There it is. Now I need to figure out why nothing else is happening.  (You can skip the middle line from now on)

**Comment by mackworth** [comment]
Let’s try: 
Bash -x compressor/compressorBatch.sh -i journey.mpg journey.MP4

**Comment by mackworth** [comment]
Sorry: bash, not Bash

**Comment by tdlivings** [comment]
thomaslivings@Thomas-MacBook-Pro ~ % cd downloads
thomaslivings@Thomas-MacBook-Pro downloads % bash -x compressor/compressorBatch.sh -i journey.mpg journey.MP4
+ echo 'Version 0.1'
Version 0.1
+ compressor_dir=/Applications
+ settings='TiVo HD'
+ declare -a opts_pre_input opts_post_input
+ ((  3 > 0  ))
+ [[ -i == \-\h ]]
+ [[ -i == \-\i ]]
+ input=journey.mpg
+ shift 1
+ shift
+ ((  1 > 0  ))
+ [[ journey.MP4 == \-\h ]]
+ [[ journey.MP4 == \-\i ]]
+ [[ journey.MP4 == \-\s\e\t\t\i\n\g\s ]]
+ [[ journey.MP4 == \-\a\p\p\d\i\r ]]
+ ((  1 == 1  ))
+ output=journey.MP4
+ shift
+ ((  0 > 0  ))
+ compressor_dir=/Applications
+ compressor_path=/Applications/Compressor.app/Contents/MacOS/Compressor
+ settings_dir='~/Library/Application Support/Compressor/Settings/'
+ settings_dir='/Users/thomaslivings/Library/Application Support/Compressor/Settings/'
+ settings_path='/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting'
+ echo '
Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      journey.mpg
Output file:     journey.MP4
'

Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      journey.mpg
Output file:     journey.MP4

+ [[ -z journey.mpg ]]
+ [[ -z journey.MP4 ]]
+ [[ ! -x /Applications/Compressor.app/Contents/MacOS/Compressor ]]
+ [[ ! -f /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting ]]
+ newline='\n'
++ /Applications/Compressor.app/Contents/MacOS/Compressor -jobpath '"journey.mpg"' -settingpath '"/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting"' -locationpath '"journey.MP4"'

**Comment by tdlivings** [comment]
no joy

**Comment by mackworth** [comment]
Ok, let's try brute force:

`/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath '"journey.mpg"' -settingpath '"/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting"' -locationpath '"journey.MP4"'`

and if that doesn't work, try:
`/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "journey.MP4"`

**Comment by mackworth** [comment]
And if neither of those, how about 
`/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4"`

**Comment by tdlivings** [comment]
the third one inserted a batch in compressor

**Comment by tdlivings** [comment]
encoding now

**Comment by tdlivings** [comment]
journey.mpg recoded to journey.m4v successfully

**Comment by tdlivings** [comment]
![Uploading Screen Shot 2020-11-18 at 10.54.30 PM.png…]()

**Comment by mackworth** [comment]
It stripped your screenshot out.  
There should have been some text returned from compressor when it successfully launched?

So, now we know: compressor api doesn't let you use relative pathnames (and helpfully gives NO error message). 

I'll give you a batch file that converts them to absolute ones, but that'll be tomorrow.

**Comment by tdlivings** [comment]
Thank you

**Comment by mackworth** [comment]
And was there any text sent back?

**Comment by tdlivings** [comment]
where would the text be?

**Comment by tdlivings** [comment]
this is the only message after running the script:

:CMedia asking for audio track at index 0, base class supports only 1 or zero tracks
2020-11-18 22:50:23.916 Compressor[1638:710386] owner is: thomaslivings
2020-11-18 22:50:23.931 Compressor[1638:710386] <jobID 37FDFACD-5802-4102-A07E-83AD9A2F29F1 /> <batchID 5674DC9B-FAFC-453B-980C-752D47850F77 />

**Comment by mackworth** [comment]
That's exactly what I was looking for: the jobID.
Give this one a try. First try it with the same command line: 

````
cd ~/Movies
<<wherever batch file is>>/compressorBatch.sh -i journey.mpg journey.MP4
````
Note that this will say "Version 0.2"

If that works, then you can try updating your Compressor Format to point to this shell script and try it again in cTiVo.
[compressorBatch.sh.zip](https://github.com/mackworth/cTiVo/files/5569079/compressorBatch.sh.zip)

**Comment by tdlivings** [comment]
I am sorry I am the worse tester ever. I am back offline for 2 days.

**Comment by mackworth** [comment]
No problem at all. This is an unusual process, as I can't replicate the environment, but the ability to add a networkable encoding system is pretty cool.

**Comment by tdlivings** [comment]
Are we doing terminal shells still, Thank You

**Comment by tdlivings** [comment]
Last login: Fri Nov 20 19:54:32 on ttys000
thomaslivings@Thomas-MacBook-Pro ~ % chmod+x cd ~/Movies
/Users/thomaslivings/Downloads/compressor/compressorBatch.sh -i journey.mpg journey.MP4

zsh: command not found: chmod+x
Version 0.2
+ declare -a opts_pre_input opts_post_input
+ ((  3 > 0  ))
+ [[ -i == \-\h ]]
+ [[ -i == \-\i ]]
+ input=journey.mpg
+ [[ ! journey.mpg == /* ]]
++ PWD
+ input=/Users/thomaslivings/journey.mpg
+ shift 1
+ shift
+ ((  1 > 0  ))
+ [[ journey.MP4 == \-\h ]]
+ [[ journey.MP4 == \-\i ]]
+ [[ journey.MP4 == \-\s\e\t\t\i\n\g\s ]]
+ [[ journey.MP4 == \-\a\p\p\d\i\r ]]
+ ((  1 == 1  ))
+ output=journey.MP4
+ [[ ! journey.MP4 == /* ]]
++ PWD
+ output=/Users/thomaslivings/journey.MP4
+ shift
+ ((  0 > 0  ))
+ compressor_dir=/Applications
+ compressor_path=/Applications/Compressor.app/Contents/MacOS/Compressor
+ settings_dir='~/Library/Application Support/Compressor/Settings/'
+ settings_dir='/Users/thomaslivings/Library/Application Support/Compressor/Settings/'
+ settings_path='/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting'
+ echo '
Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      /Users/thomaslivings/journey.mpg
Output file:     /Users/thomaslivings/journey.MP4
'

Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      /Users/thomaslivings/journey.mpg
Output file:     /Users/thomaslivings/journey.MP4

+ [[ -z /Users/thomaslivings/journey.mpg ]]
+ [[ ! -f /Users/thomaslivings/journey.mpg ]]
+ [[ -z /Users/thomaslivings/journey.MP4 ]]
+ [[ ! -x /Applications/Compressor.app/Contents/MacOS/Compressor ]]
+ [[ ! -f /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting ]]
+ newline='\n'
++ /Applications/Compressor.app/Contents/MacOS/Compressor -jobpath /Users/thomaslivings/journey.mpg -settingpath '/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting' -locationpath /Users/thomaslivings/journey.MP4

**Comment by tdlivings** [comment]
successful batch

**Comment by mackworth** [comment]
Good, although I’m not seeing the last but of output, which is odd. Can you try it in cTiVo now? Just make sure your Compressor Format is calling this compBatch.sh

**Comment by tdlivings** [comment]
Houston..... we have lift off

**Comment by tdlivings** [comment]
![Uploading Screen Shot 2020-11-20 at 8.49.49 PM.png…]()

**Comment by tdlivings** [comment]
<img width="367" alt="Screen Shot 2020-11-20 at 8 50 20 PM" src="https://user-images.githubusercontent.com/22117759/99867836-0aa84c80-2b72-11eb-9f3c-cd91cd23a360.png">

**Comment by tdlivings** [comment]
uh oh, it downloaded and encoded again, even though it succeeded the first time. Was it supposed to run twice?

**Comment by tdlivings** [comment]
[com.cTiVo.cTiVo 2020-11-21--04-40-23-326.log](https://github.com/mackworth/cTiVo/files/5577026/com.cTiVo.cTiVo.2020-11-21--04-40-23-326.log)

**Comment by tdlivings** [comment]
https://youtu.be/Eu-xU772Glo

**Comment by mackworth** [comment]
Good progress, but we're not getting anything back from the submittal process. Thus while monitoring the encoder, cTiVo thinks it's failed and goes through the download process again.  You can stop it obviously.

I'm trying to figure out why the script stops running after the submittal.  Let's try this one:
Just use it from Terminal as before (should still have video file in same place, but delete output video file before running). Let me know what output says. (note it should also now give you a better job name in Compressor) 

[compressorBatch.sh 2.1.zip](https://github.com/mackworth/cTiVo/files/5577066/compressorBatch.sh.2.1.zip)

**Comment by mackworth** [comment]
And then try copy/pasting this into Terminal as well:
````
rm /Users/thomaslivings/journey.MP4
/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath /Users/thomaslivings/journey.mpg -settingpath '/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting' -locationpath /Users/thomaslivings/journey.MP4 2>&1
````

**Comment by tdlivings** [comment]
ok, I am a bit confused. Should I put a mpg file called journey in the downloads folder again?

**Comment by mackworth** [comment]
Sorry, yes.  

then 
```
cd ~/Downloads
rm journey.MP4
```
then
`<<wherever batch file is>>/compressorBatch.sh -i journey.mpg journey.MP4`

Then 
`/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4" `

**Comment by tdlivings** [comment]
nothing on the terminal/compressorbatch.sh (probably user error like a mpg not where it was supposed to be

second script worked well

**Comment by mackworth** [comment]
Any text output?

**Comment by tdlivings** [comment]
rm /Users/thomaslivings/journey.MP4
/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath /Users/thomaslivings/journey.mpg -settingpath '/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting' -locationpath /Users/thomaslivings/journey.MP4 2>&1
rm: /Users/thomaslivings/journey.MP4: No such file or directory
CVideoUtilities::CMedia asking for audio track at index 0, base class supports only 1 or zero tracks
2020-11-20 21:29:38.382 Compressor[3213:1267299] owner is: thomaslivings
2020-11-20 21:29:38.400 Compressor[3213:1267299] <jobID 85B94C11-A3F0-4F8E-B091-6AE6CA1EE979 /> <batchID 62AEBE4F-9303-4387-A738-D839377D25BF />

**Comment by mackworth** [comment]
We're out of sync (I think).  My second group was slightly different that the first group.

Can you do these:

```
cd ~/Downloads
rm journey.MP4
```
then this (assuming batch file is in Downloads/Compressor/ directory)
`Compressor/compressorBatch.sh -i journey.mpg journey.MP4`

Then 
`/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4" `

Assuming the batch fails, and the last line worked, then cut/paste the following, but then cut/paste the JOBIDHERE part with whatever the long jobid code is (e.g. 85B94C11-A3F0-4F8E-B091-6AE6CA1EE979 above)

`/Applications/Compressor.app/Contents/MacOS/Compressor -monitor -jobid JOBIDHERE

**Comment by tdlivings** [comment]
cd ~/Downloads
rm journey.mp4
/Users/thomaslivings/Downloads/compressor/compressorBatch.sh -i journey.mpg journey.MP4
rm: journey.mp4: No such file or directory
Version 0.21
+ declare -a opts_pre_input opts_post_input
+ ((  3 > 0  ))
+ [[ -i == \-\h ]]
+ [[ -i == \-\i ]]
+ input=journey.mpg
+ [[ ! journey.mpg == /* ]]
++ PWD
+ input=/Users/thomaslivings/Downloads/journey.mpg
+ shift 1
+ shift
+ ((  1 > 0  ))
+ [[ journey.MP4 == \-\h ]]
+ [[ journey.MP4 == \-\i ]]
+ [[ journey.MP4 == \-\s\e\t\t\i\n\g\s ]]
+ [[ journey.MP4 == \-\a\p\p\d\i\r ]]
+ ((  1 == 1  ))
+ output=journey.MP4
+ [[ ! journey.MP4 == /* ]]
++ PWD
+ output=/Users/thomaslivings/Downloads/journey.MP4
+ shift
+ ((  0 > 0  ))
+ compressor_dir=/Applications
+ compressor_path=/Applications/Compressor.app/Contents/MacOS/Compressor
+ settings_dir='~/Library/Application Support/Compressor/Settings/'
+ settings_dir='/Users/thomaslivings/Library/Application Support/Compressor/Settings/'
+ settings_path='/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting'
++ basename /Users/thomaslivings/Downloads/journey.mpg
+ name=journey.mpg
+ echo '
Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      /Users/thomaslivings/Downloads/journey.mpg
Output file:     /Users/thomaslivings/Downloads/journey.MP4
'

Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      /Users/thomaslivings/Downloads/journey.mpg
Output file:     /Users/thomaslivings/Downloads/journey.MP4

+ [[ -z /Users/thomaslivings/Downloads/journey.mpg ]]
+ [[ ! -f /Users/thomaslivings/Downloads/journey.mpg ]]
+ echo 'Error: input file /Users/thomaslivings/Downloads/journey.mpg not found'
Error: input file /Users/thomaslivings/Downloads/journey.mpg not found
+ usage
+ cat

Script to drive compressor

Usage: /Users/thomaslivings/Downloads/compressor/compressorBatch.sh -i INPUTVIDEONAME [-settings "SETTINGSNAME"] [-appdir "APPLICATIONDIR"] <compressor options> OUTPUTFILENAME
       /Users/thomaslivings/Downloads/compressor/compressorBatch.sh -h

To specify compressor settings, just add 
	-settings and the name of the settings (use quotes or / before spaces) 
  Default settings is named "TiVo HD".

If compressor is NOT in /Applications, then add 
	-appdir "DirectoryWhereCompressorIsStored"

For this text, use the "-h" option

Other parameters will be passed through from calling program.

+ exit 1
thomaslivings@Thomas-MacBook-Pro Downloads %

**Comment by tdlivings** [comment]
ok hold on something deleted the journey.mpg. let me recreate it.

**Comment by mackworth** [comment]
That says that `Users/thomaslivings/Downloads/journey.mpg not found`. Do you need to copy it from ~/journey.mpg to the Downloads dir?

mv ~/journey.mpg ~/Downloads/journey.mpg

**Comment by tdlivings** [comment]
Done, running scripts wait one

**Comment by tdlivings** [comment]
Last login: Fri Nov 20 21:41:37 on ttys000
thomaslivings@Thomas-MacBook-Pro ~ % cd ~/Downloads
rm journey.MP4
Compressor/compressorBatch.sh -i journey.mpg journey.MP4
rm: journey.MP4: No such file or directory
Version 0.21
+ declare -a opts_pre_input opts_post_input
+ ((  3 > 0  ))
+ [[ -i == \-\h ]]
+ [[ -i == \-\i ]]
+ input=journey.mpg
+ [[ ! journey.mpg == /* ]]
++ PWD
+ input=/Users/thomaslivings/Downloads/journey.mpg
+ shift 1
+ shift
+ ((  1 > 0  ))
+ [[ journey.MP4 == \-\h ]]
+ [[ journey.MP4 == \-\i ]]
+ [[ journey.MP4 == \-\s\e\t\t\i\n\g\s ]]
+ [[ journey.MP4 == \-\a\p\p\d\i\r ]]
+ ((  1 == 1  ))
+ output=journey.MP4
+ [[ ! journey.MP4 == /* ]]
++ PWD
+ output=/Users/thomaslivings/Downloads/journey.MP4
+ shift
+ ((  0 > 0  ))
+ compressor_dir=/Applications
+ compressor_path=/Applications/Compressor.app/Contents/MacOS/Compressor
+ settings_dir='~/Library/Application Support/Compressor/Settings/'
+ settings_dir='/Users/thomaslivings/Library/Application Support/Compressor/Settings/'
+ settings_path='/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting'
++ basename /Users/thomaslivings/Downloads/journey.mpg
+ name=journey.mpg
+ echo '
Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      /Users/thomaslivings/Downloads/journey.mpg
Output file:     /Users/thomaslivings/Downloads/journey.MP4
'

Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      /Users/thomaslivings/Downloads/journey.mpg
Output file:     /Users/thomaslivings/Downloads/journey.MP4

+ [[ -z /Users/thomaslivings/Downloads/journey.mpg ]]
+ [[ ! -f /Users/thomaslivings/Downloads/journey.mpg ]]
+ [[ -z /Users/thomaslivings/Downloads/journey.MP4 ]]
+ [[ ! -x /Applications/Compressor.app/Contents/MacOS/Compressor ]]
+ [[ ! -f /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting ]]
+ newline='\n'
++ /Applications/Compressor.app/Contents/MacOS/Compressor -batchname 'cTiVo: ' -jobpath /Users/thomaslivings/Downloads/journey.mpg -settingpath '/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting' -locationpath /Users/thomaslivings/Downloads/journey.MP4

**Comment by tdlivings** [comment]
here is what I ran: cd ~/Downloads
rm journey.MP4
Compressor/compressorBatch.sh -i journey.mpg journey.MP4

**Comment by mackworth** [comment]
Yup and it's looking great, except that nothing comes back from the call to Compressor, and nothing gets executed after that call.  Weird.

Can you try this part:

`/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4" `

Then cut/paste the following, but then cut/paste the JOBIDHERE part with whatever the long jobid code is (e.g. 85B94C11-A3F0-4F8E-B091-6AE6CA1EE979, but whatever it actually says)

`/Applications/Compressor.app/Contents/MacOS/Compressor -monitor -jobid JOBIDHERE

**Comment by tdlivings** [comment]
here is the output results:
Last login: Fri Nov 20 21:47:58 on ttys001
thomaslivings@Thomas-MacBook-Pro ~ % cd ~/Downloads
rm journey.MP4
Compressor/compressorBatch.sh -i journey.mpg journey.MP4
rm: journey.MP4: No such file or directory
Version 0.21
+ declare -a opts_pre_input opts_post_input
+ ((  3 > 0  ))
+ [[ -i == \-\h ]]
+ [[ -i == \-\i ]]
+ input=journey.mpg
+ [[ ! journey.mpg == /* ]]
++ PWD
+ input=/Users/thomaslivings/Downloads/journey.mpg
+ shift 1
+ shift
+ ((  1 > 0  ))
+ [[ journey.MP4 == \-\h ]]
+ [[ journey.MP4 == \-\i ]]
+ [[ journey.MP4 == \-\s\e\t\t\i\n\g\s ]]
+ [[ journey.MP4 == \-\a\p\p\d\i\r ]]
+ ((  1 == 1  ))
+ output=journey.MP4
+ [[ ! journey.MP4 == /* ]]
++ PWD
+ output=/Users/thomaslivings/Downloads/journey.MP4
+ shift
+ ((  0 > 0  ))
+ compressor_dir=/Applications
+ compressor_path=/Applications/Compressor.app/Contents/MacOS/Compressor
+ settings_dir='~/Library/Application Support/Compressor/Settings/'
+ settings_dir='/Users/thomaslivings/Library/Application Support/Compressor/Settings/'
+ settings_path='/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting'
++ basename /Users/thomaslivings/Downloads/journey.mpg
+ name=journey.mpg
+ echo '
Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      /Users/thomaslivings/Downloads/journey.mpg
Output file:     /Users/thomaslivings/Downloads/journey.MP4
'

Compressor Path: /Applications/Compressor.app/Contents/MacOS/Compressor 
Settings Path:   /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting
Input file:      /Users/thomaslivings/Downloads/journey.mpg
Output file:     /Users/thomaslivings/Downloads/journey.MP4

+ [[ -z /Users/thomaslivings/Downloads/journey.mpg ]]
+ [[ ! -f /Users/thomaslivings/Downloads/journey.mpg ]]
+ [[ -z /Users/thomaslivings/Downloads/journey.MP4 ]]
+ [[ ! -x /Applications/Compressor.app/Contents/MacOS/Compressor ]]
+ [[ ! -f /Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting ]]
+ newline='\n'
++ /Applications/Compressor.app/Contents/MacOS/Compressor -batchname 'cTiVo: ' -jobpath /Users/thomaslivings/Downloads/journey.mpg -settingpath '/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting' -locationpath /Users/thomaslivings/Downloads/journey.MP4

**Comment by mackworth** [comment]
Oh, and if that works and starts spewing information, then in another terminal session use the same jobID with

`/Applications/Compressor.app/Contents/MacOS/Compressor -kill -jobid JOBIDHERE

**Comment by tdlivings** [comment]
I am not seeing a job id in the results this time

**Comment by mackworth** [comment]
have to use the direct call to Compressor:

`/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4" `

Then cut/paste the following, but then replace the JOBIDHERE part with the jobid code is (e.g. 85B94C11-A3F0-4F8E-B091-6AE6CA1EE979, but whatever it actually says)

`/Applications/Compressor.app/Contents/MacOS/Compressor -monitor -jobid JOBIDHERE`

then in another terminal session use the same jobID with

`/Applications/Compressor.app/Contents/MacOS/Compressor -kill -jobid JOBIDHERE`

**Comment by tdlivings** [comment]
here is what is ran:

rm /Users/thomaslivings/journey.MP4
/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath /Users/thomaslivings/journey.mpg -settingpath '/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting' -locationpath /Users/thomaslivings/journey.MP4 2>&1

**Comment by tdlivings** [comment]
here is the output:

Last login: Fri Nov 20 21:48:07 on ttys000
thomaslivings@Thomas-MacBook-Pro ~ % rm /Users/thomaslivings/journey.MP4
/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath /Users/thomaslivings/journey.mpg -settingpath '/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting' -locationpath /Users/thomaslivings/journey.MP4 2>&1

rm: /Users/thomaslivings/journey.MP4: No such file or directory
CVideoUtilities::CMedia asking for audio track at index 0, base class supports only 1 or zero tracks
2020-11-20 21:57:19.331 Compressor[3511:1289613] owner is: thomaslivings
2020-11-20 21:57:19.351 Compressor[3511:1289613] <jobID 416D8471-97E0-4372-870E-89A8FE236707 /> <batchID C0596FF6-41F3-47AF-B9C3-902E7177847E />

**Comment by tdlivings** [comment]
no job created

**Comment by mackworth** [comment]
Wait. Is the movie in  /Users/thomaslivings/ or  /Users/thomaslivings/Downloads?
This assumes Downloads:
````
/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4"
````

**Comment by tdlivings** [comment]
here is what I ran:

/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath “/Users/thomaslivings/Downloads/journey.MP4"

`/Applications/Compressor.app/Contents/MacOS/Compressor -monitor -jobid 416D8471-97E0-4372-870E-89A8FE236707

**Comment by tdlivings** [comment]
here is the output:

dquote bquote>

**Comment by tdlivings** [comment]
I must be doing something wrong now

**Comment by mackworth** [comment]
it's the curly quote on the last section.  sorry.
`/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4"`

**Comment by mackworth** [comment]
The shell is picky.

**Comment by tdlivings** [comment]
this works:

cd ~/Downloads
rm journey.MP4
Compressor/compressorBatch.sh -i journey.mpg journey.MP4

this works too:

/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4"

Output from second command:

Last login: Fri Nov 20 22:07:23 on ttys000
thomaslivings@Thomas-MacBook-Pro ~ % /Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4"
CVideoUtilities::CMedia asking for audio track at index 0, base class supports only 1 or zero tracks
2020-11-20 22:09:25.197 Compressor[3627:1297317] owner is: thomaslivings
2020-11-20 22:09:25.211 Compressor[3627:1297317] <jobID 200DCEBD-91B7-4F8A-8776-839C0EBA145F /> <batchID ABE4F2EE-598E-4250-A666-2B5946229870 />
thomaslivings@Thomas-MacBook-Pro ~ %

**Comment by mackworth** [comment]
ok and now:
/Applications/Compressor.app/Contents/MacOS/Compressor -monitor -jobid 200DCEBD-91B7-4F8A-8776-839C0EBA145F

then after that runs for a bit, type in another terminal session (CMD-N) use the same jobID with

/Applications/Compressor.app/Contents/MacOS/Compressor -kill -jobid 200DCEBD-91B7-4F8A-8776-839C0EBA145F

**Comment by tdlivings** [comment]
`/Applications/Compressor.app/Contents/MacOS/Compressor -monitor -jobid 200DCEBD-91B7-4F8A-8776-839C0EBA145F doesn't work (bquote)

**Comment by mackworth** [comment]
Take off the first `

And after all that,  one last bit of info for me, before i call it a night:  After you run the batch 

Compressor/compressorBatch.sh -i journey.mpg journey.MP4
type
`echo $?`

**Comment by tdlivings** [comment]
ran:

/Applications/Compressor.app/Contents/MacOS/Compressor -monitor -jobid 200DCEBD-91B7-4F8A-8776-839C0EBA145F

failed to generate a batch

**Comment by mackworth** [comment]
it's not supposed to; it's supposed to give continuous output on the status of the existing job

**Comment by mackworth** [comment]
hence the -monitor part.  It's supposed to give continuous output, then it's supposed to exit once the job is finished.  FYI, the other command is supposed to be able to kill the job once it's created, in case the user cancels it, or quits the program, or there's other issues.

**Comment by tdlivings** [comment]
Here is what I ran:

/Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath “/Users/thomaslivings/Downloads/journey.MP4"

/Applications/Compressor.app/Contents/MacOS/Compressor -monitor -jobid C24BEA92-7246-47D0-AFA2-D599C056EB74

/Applications/Compressor.app/Contents/MacOS/Compressor -kill -jobid C24BEA92-7246-47D0-AFA2-D599C056EB74

it created a batch and killed the batch

**Comment by tdlivings** [comment]
output results:

Last login: Fri Nov 20 22:14:00 on ttys001
thomaslivings@Thomas-MacBook-Pro ~ % /Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4"


CVideoUtilities::CMedia asking for audio track at index 0, base class supports only 1 or zero tracks
2020-11-20 22:16:55.040 Compressor[3676:1303433] owner is: thomaslivings
2020-11-20 22:16:55.055 Compressor[3676:1303433] <jobID C24BEA92-7246-47D0-AFA2-D599C056EB74 /> <batchID CF4DA775-3C5C-4153-90EF-00ABDEFFE4FD />
thomaslivings@Thomas-MacBook-Pro ~ % AVAFRenderer AVAFPluginConnect name : AVAGetRendererInfo 
 AVAFRenderer AVAFPluginConnect name : AVACreateRenderer 
 AVAFRenderer AVAFPluginConnect name : AVACreateRendererDisplayExt 
 AVAFRenderer AVAFPluginConnect name : AVACreateRendererDVDExt 
thomaslivings@Thomas-MacBook-Pro ~ % /Applications/Compressor.app/Contents/MacOS/Compressor -monitor -jobid C24BEA92-7246-47D0-AFA2-D599C056EB74
<jobStatus name="Journey.mpg" submissionTime="11/20/20, 10:16:55 PM" sentBy="thomaslivings" jobType="" priority="HighPriority" timeElapsed="1 minute(s)" timeRemaining="2 minute(s)" timeElapsedSeconds="82" timeRemainingSeconds="136" percentComplete="37" resumePercentComplete="0" status="Processing" jobid="C24BEA92-7246-47D0-AFA2-D599C056EB74" batchid="CF4DA775-3C5C-4153-90EF-00ABDEFFE4FD" /jobStatus>
thomaslivings@Thomas-MacBook-Pro ~ % /Applications/Compressor.app/Contents/MacOS/Compressor -kill -jobid C24BEA92-7246-47D0-AFA2-D599C056EB74

thomaslivings@Thomas-MacBook-Pro ~ % echo $?
0
thomaslivings@Thomas-MacBook-Pro ~ %

**Comment by mackworth** [comment]
Can you try anoterh submittal again adding `-query 2 -timeout 10`?

````
rm /Users/thomaslivings/Downloads/journey.MP4
 /Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4"

`/Applications/Compressor.app/Contents/MacOS/Compressor -monitor -query 2 -timeout 10 -jobid JOBID
````

And after all that:

````
Compressor/compressorBatch.sh -i journey.mpg journey.MP4
echo $?
````

**Comment by tdlivings** [comment]
Here is what I ran:

rm /Users/thomaslivings/Downloads/journey.MP4
 /Applications/Compressor.app/Contents/MacOS/Compressor -jobpath "/Users/thomaslivings/Downloads/journey.mpg" -settingpath "/Users/thomaslivings/Library/Application Support/Compressor/Settings/TiVo HD.compressorsetting" -locationpath "/Users/thomaslivings/Downloads/journey.MP4"

`/Applications/Compressor.app/Contents/MacOS/Compressor -monitor -query 2 -timeout 10 -jobid JOBID
And after all that:

Compressor/compressorBatch.sh -i journey.mpg journey.MP4
echo $?

resulted in 5 batches created

**Comment by tdlivings** [comment]
`/Applications/Compressor.app/Contents/MacOS/Compressor -monitor -query 2 -timeout 10 -jobid JOBID
rm: /Users/thomaslivings/Downloads/journey.MP4: No such file or directory
CVideoUtilities::CMedia asking for audio track at index 0, base class supports only 1 or zero tracks
2020-11-20 22:33:52.869 Compressor[3804:1316219] owner is: thomaslivings
2020-11-20 22:33:52.884 Compressor[3804:1316219] <jobID 25D62A5D-8B7E-4B52-A605-027FAE0099FB /> <batchID 73230C69-5145-4028-972F-F168AB6BD012 />
rm: /Users/thomaslivings/Downloads/journey.MP4: No such file or directory
CVideoUtilities::CMedia asking for audio track at index 0, base class supports only 1 or zero tracks
2020-11-20 22:33:56.362 Compressor[3812:1316439] owner is: thomaslivings
2020-11-20 22:33:56.380 Compressor[3812:1316439] <jobID A9F45705-275A-452D-B6DC-3DD6BE57D1FC /> <batchID 700B726F-94AE-4EFF-A4AA-D6B20E03D212 />
rm: /Users/thomaslivings/Downloads/journey.MP4: No such file or directory
CVideoUtilities::CMedia asking for audio track at index 0, base class supports only 1 or zero tracks
2020-11-20 22:33:59.723 Compressor[3821:1316651] owner is: thomaslivings
2020-11-20 22:33:59.742 Compressor[3821:1316651] <jobID E4B29D3F-D030-41C4-90F2-9EF9647FC562 /> <batchID 1701D5AF-DCD3-4E58-9834-993BBDE8DABE />
rm: /Users/thomaslivings/Downloads/journey.MP4: No such file or directory
CVideoUtilities::CMedia asking for audio track at index 0, base class supports only 1 or zero tracks
2020-11-20 22:34:03.102 Compressor[3825:1316764] owner is: thomaslivings
2020-11-20 22:34:03.122 Compressor[3825:1316764] <jobID 43D90A07-74B5-4261-A813-566D80C39BB6 /> <batchID 21797DEC-DCBE-41F8-9DBC-80BDA0173505 />
rm: /Users/thomaslivings/Downloads/journey.MP4: No such file or directory
CVideoUtilities::CMedia asking for audio track at index 0, base class supports only 1 or zero tracks
2020-11-20 22:34:06.490 Compressor[3808:1316905] owner is: thomaslivings
2020-11-20 22:34:06.510 Compressor[3808:1316905] <jobID 889A9488-13D1-434F-9B4D-05FA15DBC9B4 /> <batchID B6B75105-CBDD-496C-9B70-87FFACBF110B />
Compressor/compressorBatch.sh -i journey.mpg journey.MP4
echo $?

**Comment by tdlivings** [comment]
Probably me....

**Comment by mackworth** [comment]
Wel, I'm gonna have to do some research.  I'll get back to you.   

Thx!

**Comment by tdlivings** [comment]
ok, no problem I am pretty much done anyway

**Comment by mackworth** [comment]
So, I experimented with a few things that could cause the symptoms we were seeing, then realized something must be wrong with the API. I looked at Compressor in the App Store, realized it was cheaper than I thought, so went ahead and bought it. 

Good thing, too. I’ve found several serious issues that would have been very hard to diagnose remotely. I’ve gotten through a few of them, but have a ways to go. Will get back to you.

**Comment by tdlivings** [comment]
I am glad the hear of your progress. I look forward to seeing your solutions.

Happy Thanksgiving

**Comment by mackworth** [comment]
Well, here you go.  Give this a try. The Format works, but I'm not happy at all with Compressor.

There's a new Format here, so I suggest you delete the existing one from cTiVo before even opening this one to avoid confusion. The script assumes that your cTiVo is in /Applications.

I've found many bugs in Compressor in this process, and am filing reports. Notably, it will not accept a Transport Stream file, very common in video world. I've gotten around that by copying it with ffmpeg to an MP4 temp file, which is quite fast (no transcoding, just copying bitstreams), but still ugly. Secondly, it seems very sensitive to video issues in the file. In my recent batch of 20 files, it failed about half of the H.264 files. I've also seen it leave instances of Compressor chewing up CPU time in the background with 

Assuming you see the same, you might try downloading them with cTIVo in the Decrypt H.264 Format (which does the same copying to disk) and then manually submitting them to Compressor to confirm it does the same from the UI.

Note that if you submit more jobs to the Compressor (using Simultaneous Encoders in Adv Pref) than it can handle at once (even two), then it will timeout in cTiVo, which will assume it has frozen. I've hacked my copy of cTIVo to fix this, and have a way I can do it correctly, but that depends if this is worth it.

[Compressor0.3.zip](https://github.com/mackworth/cTiVo/files/5593496/Compressor0.3.zip)

**Comment by mackworth** [comment]
oh, you could try some different Presets in Compressor as well; that may have an effect. Just add `[-settings "SETTINGSNAME"]` to the Video Options.  At the moment, it can't call any built-in presets, you'll have to duplicate them as custom.

**Comment by tdlivings** [comment]
Thank you for the files. I will give them a try very soon and get back to you.

**Comment by tdlivings** [comment]
[com.cTiVo.cTiVo 2020-11-26--04-19-43-845.log](https://github.com/mackworth/cTiVo/files/5600418/com.cTiVo.cTiVo.2020-11-26--04-19-43-845.log)


no joy...

**Comment by tdlivings** [comment]
When I asked for the enhancement I was trying to get the use of more GPU resources while pulling down the TiVo files. Compressor seemed like a good option to make that happen. In hindsight I see that it is probably not a good option for the public at large. First is a dollar cost to get it, where as handbrake is free. Second is the complexity of Compressor. I am not surprised abut the *.ts files. Compressor didn't play well with the TiVo decrypted files until recently and Final cut will not even recognize them. Hence the need to encode out of the box. 

For now, perhaps, it will be best to simply include this enhancement in the advance options of the toolkit?

Thank you for all of your hard work on the subject. I will continue to test it out.

**Comment by mackworth** [comment]
Any reactions/results so far?

**Comment by mackworth** [comment]
Oh, btw, if you want to download a file into Final Cut, and you know a channel is H.264, then download files with the Decrypt MP4 Format. It will do the same copying process from .TS file to an MP4, no transcoding, and it should load into Final Cut.

**Comment by tdlivings** [comment]
I use OTA signals. I wish I could use the mp4 transmission of the cable companies but I don’t want to pay for the service. That is why I have such a challenge getting an editable format. Final Cut does not want mpg files. I was using ProRes LT from CTiVo but there is a sound sync issue 25% of the time. 

Now that you fixed the handbrake options I can use that flawlessly and also the compressor import features so I can use GPU resources because handbrake is only CPU for the h.264 encodes. 

Happy thanksgiving

**Comment by mackworth** [comment]
That’s interesting. They’re not using any H.264 OTA?  That would suggest you’d get a better result with the Compressor Format than I did, as most of my failures were H.264.

**Comment by tdlivings** [comment]
yes, all OTA is coming in the TiVo as mpg (non-editable in the apple world). This is definitely a TiVo choice but I have an older box Cir. 2015. I have some choices on the pull down thanks to cTiVo. I can't see using CPU resources for the pull down just to edit and Use more resources to encode the edited product. I supposed it is just my personal preference but I am worried about burning up a 3600 dollar machine. If that is the case I can merely purchase the content and save the hardware. My current workflow is pull as HB / Compressor > FCPX > output as m4v. This seems to be the most efficient use of my hardware kit.

I was testing on a MacBook Pro 8 core but have cut over to my main iMac Pro with an eGPu (Blackmagic). My only concern is sound and video sync in compressor. I will have more later this week.

**Comment by tdlivings** [comment]
It is important to note, when I import the compressor.plist into cTiVo it will not show up in the available profiles unless I duplicate it. 

After that it is available in the drop down when "show in user interface" is selected.

see video: https://youtu.be/OOmDsPx_fFk

It worked flawlessly other than that and it uses GPU not CPU for compressor.

<img width="1346" alt="Screen Shot 2020-11-30 at 11 59 08" src="https://user-images.githubusercontent.com/22117759/100658221-8db06c00-3303-11eb-96b3-c4ac2bc1a5d9.png">

**Comment by tdlivings** [comment]
Ran a test of the concurrent compressor job function that would emulate the subscription pull downs in cTiVo, while using compressor profile. As you predicted the second job fails because the first one is still running. Also seeing "File too short" even though the job completes. This mainly happens on SD shows. 

[com.cTiVo.cTiVo 2020-12-02--06-33-33-912.log](https://github.com/mackworth/cTiVo/files/5629735/com.cTiVo.cTiVo.2020-12-02--06-33-33-912.log)

**Comment by mackworth** [comment]
Same problem as well, I think.  

Here's the experimental cTiVo version I mentioned. It recognizes Formats with names starting with `Compressor`, and will never time out on them.  Should avoid the `too short` message as well. You can tell that it's running as `About cTiVo` will say `Version 3.4.4-Compressor (1127)` Should allow you to enter multiple Compressor jobs at once, and manage them from the Compressor queue. If you delete it there, it should then fail it in cTiVo as well.

https://www.dropbox.com/s/gum0t9ps06irbpq/cTiVoCompressor.zip?dl=0

**Comment by tdlivings** [comment]
Got it. I will check it out shortly.

**Comment by tdlivings** [comment]
sorry I have not gotten back to you sooner. I am running into a problem with compressor and Final Cut in the area of the audio track management. This issue is not related to your efforts. For some reason compressor has a problem with the multi track steams like dolby 5.1. Then when I take it into Final cut I get the sound effect channel but the dialogue is silent. I have to do some tweaking on my TiVo HD profile before I can continue testing the new ctivo.

Thank you

**Comment by mackworth** [comment]
Thanks for update. Have you tried running multiple shows simultaneously? Change num encoders in Adv prefs, and you should be able to download them all and feed them into Compressor.

**Comment by tdlivings** [comment]
Thank you. Will do

**Comment by tdlivings** [comment]
I created two 10 minute recordings this morning. I will try the multi encoder on Saturday and upload the log files.

Thank you

**Comment by tdlivings** [comment]
Number of concurrent encoders at 2, two 3 minute recording successfully processed. I am trying two one hour shows now. I stopped the two and changed to a one on two different boxes to truly get the concurrent encode test.

Second show failed.
[com.cTiVo.cTiVo 2020-12-10--17-16-27-519.log](https://github.com/mackworth/cTiVo/files/5674621/com.cTiVo.cTiVo.2020-12-10--17-16-27-519.log)
<img width="374" alt="Screen Shot 2020-12-10 at 09 30 42" src="https://user-images.githubusercontent.com/22117759/101807934-8db22800-3aca-11eb-8fa8-92564d60e6b2.png">

this one succeeded

**Comment by tdlivings** [comment]
More information. I did two three minute recordings that did succeed but the one was done before the two was launched. If I had the concurrent encoders set to one I am sure the process would have succeeded.

**Comment by mackworth** [comment]
The log of the one that failed suggests that it was cancelled by Compressor UI:

timeElapsed="4 minute(s)" timeRemaining="0" timeElapsedSeconds="249" timeRemainingSeconds="0" percentComplete="100" resumePercentComplete="100" status="Cancelled"

**Comment by tdlivings** [comment]
Yes I cancelled one but the second one failed.

**Comment by mackworth** [comment]
Sorry, yes; I missed it:
```
Input file:      /var/folders/vm/5qrh2ls52jxghsdn28h6t_x40000gn/T/ctivo/bufferFBI-Most Wanted - Rampage.mpg
Output file:     /Volumes/TR-04/TR04-TiVo Recordings/FBI: Most Wanted/FBI-Most Wanted - Rampage.m4v
ffmpeg:          /Applications/cTiVo.app/Contents/MacOS/ffmpeg

Launching Compressor
Complete: 0.0%
2020-12-10 09:30:58:953 -[MTTask saveLogFileType:fromPath:]@208>errFile for task encode: Error: Compressor returned no jobID!
CVideoUtilities::CMedia asking for audio track at index 0, base class supports only 1 or zero tracks
```
So that last line is what Compressor returned when we submitted that job. Sounds like it might be related to the problem you're having.

**Comment by tdlivings** [comment]
Ok, Recorded 2 tv shows on a Std. Def channel so no Dolby 5.1 on that. Show recording 14 minutes each for test purposes and on two separate boxes . Test was successful.... see log attached.
[com.cTiVo.cTiVo 2020-12-12--05-34-23-762.log](https://github.com/mackworth/cTiVo/files/5682537/com.cTiVo.cTiVo.2020-12-12--05-34-23-762.log)

**Comment by tdlivings** [comment]
Next test 4 shows on High def channels but I told the Compressor profile not to include dolby 5.1 track. two short ones (15 minutes) two that I tried previously and failed. for audio at index 0 issue.

1. HD 20/20 Partial Recording completed
2. HD Dateline Partial Recording Completed
3. HD FBI Full episode (previously failed) Completed
4. HD Good Dr Full Episode (Previously succeeded) Completed

All GPU Encoded with compressor

TiVo HD Attributes
<img width="382" alt="Screen Shot 2020-12-11 at 22 14 23" src="https://user-images.githubusercontent.com/22117759/101976872-65c3e100-3bfe-11eb-829e-643c6e3a72d0.png">

[com.cTiVo.cTiVo 2020-12-12--05-41-56-868.log](https://github.com/mackworth/cTiVo/files/5682597/com.cTiVo.cTiVo.2020-12-12--05-41-56-868.log)

**Comment by tdlivings** [comment]
I found out something in Compressor. When selecting an output profile, the user must select one that has a dolby output. If the user does not the sound track is incomplete and will be missing the voice track in most instances when a broadcast channel is using dolby 5.1. The apple devices outputs use l/r and we need C L R Ls Rs LFE.

Not your problem obviously, just thought I would throw that out there. So I changed to Video Sharing Services  > HD 1080p and selected dolby 5.1 in the audio drop down. It is working now. The only problem going forward is while using a conversion from mpg to mov, sound sync problems are inevitable. 

Your cTiVo is work though. I will continue the tests and get back to you in a few days

**Comment by tdlivings** [comment]
The new version of cTivo caused a kernel panic. I tried to find the panic log but am not able to find it in Big Sur

**Comment by mackworth** [comment]
Doing a little housecleaning, I think this issue is complete, no? If not, open it back up...

**Comment by tdlivings** [comment]
Thank you sir....
